app/models/user.rb

```diff
@@ -1,3 +1,10 @@
 class User < ApplicationRecord
-	has_many :microposts
+  has_many :microposts
+  before_save {email.downcase!}
+  validates :name, presence: true, length: {maximum: Settings.valid.name_length}
+  VALID_EMAIL_REGEX = Settings.valid.email_syntax
+  validates :email, presence: true, length: {maximum: Settings.valid.email_length},
+  format: {with: VALID_EMAIL_REGEX}, uniqueness: {case_sensitive: false}
+  has_secure_password
```

Comment:
Can you tell me here: What is has_secure_password? What is the purpose of using it?

Reply:
I read about it, it is single Rails method. It save a securely hashed "password_digest" instead of save just "password_digest" to the database; with the purpose to make the authentication more secure

Reply:
@TuanNhon Answer is not complete. Please study more about it and reply again here

Reply:
This method help us validation password
It requires to have a password_digest attribute and library BCrypt to use
  - It validate present of password
  - It validate length of password must less than or equal 72 bytes
  - It validate password_confirmation is same password_digest or not
We can choose use password_confirmation or don't use it
```
validations: false
```
I think the purpose is validate and make the authentication more secure